### DISCUSSION TITLE
Thank you for GPTPortal!

### DISCUSSION THREAD

**Discussion by VikramjitSinghRathee** [Zaki-1052/GPTPortal#15] [General]
Hi Zaki-1052, 

First, I just wanted to say that GPTPortal is really useful and simple to set up. It has proven to be very helpful to me, and I deeply appreciate the work you have done to create this repository, continue to enhance it, and share it with the public.

On another note, could you please let me know how to use the Assistants Mode? I see the button at the bottom, and clicking on it gets 'Assistants Mode ON'. I can upload documents from the top-right corner button; however, is there any specific command so that the Assistant can take up instructions and refer to the uploaded documents (ChatGPT assistants) to answer the query when 'Assistants Mode is ON'? Do both the Assistant instruction and query go into the space provided for writing the message?

**Comment by Zaki-1052** [comment]
Hello, glad you've found it helpful! For the **Assistants Mode**, the System Prompt will be populated by the contents of the [`instructions.md` file](https://github.com/Zaki-1052/GPTPortal/blob/main/public/instructions.md). When you send your query, those instructions go into the Assistant, and it starts a new "_Thread_" or conversation with the Assistant. When you upload documents with the button, it will attach those documents to the Assistant, and you can tell the Assistant to refer to those documents; it will run the Python to read it natively. If you want to refer to a previously existing Assistant (perhaps one you've created through the [OpenAI console](https://platform.openai.com/assistants), then you can copy the ID into the `.env` file where the [placeholder](https://github.com/Zaki-1052/GPTPortal/blob/d17efdd3cbc49412249edffade85d0c5ba598066/.env.example#L12) is. The same applies for if you want to continue a conversation you've been having with an Assistant in a previous session; copy the _Assistant ID_ **and** the [_Thread ID_](https://github.com/Zaki-1052/GPTPortal/blob/d17efdd3cbc49412249edffade85d0c5ba598066/.env.example#L13) into the `.env` file ([the thread can be found here](https://platform.openai.com/threads)). The Assistants API will automatically reference and reuse whatever files you had previously uploaded (either with the button or [through OpenAI](https://platform.openai.com/storage)), as well as the instructions. Those instructions can come either from you sending a message through this **GPTPortal** repo or through OpenAI's _GUI_ for making Assistants. As long as you've turned the mode on and selected an OpenAI model, then there's nothing specific to do, just upload whatever files you'd like, and the Assistant instructions will always come from that modifiable file that you can write in with any standard Text Editor. Hope this helps @VikramjitSinghRathee!

Edit: Thank you for the reminder by the way; I just realized that I hadn't finished writing the documentation for Assistants, so when I push the new Claude models later I'll add those as well! The current information under [Assistants Use](https://github.com/Zaki-1052/GPTPortal#assistants-use) is accurate, just incomplete.
